docs/architecture/adr-049-state-sync-hooks.md

```diff
@@ -22,8 +22,9 @@ the IAVL tree, they can not add their states to the snapshot stream for state-sy
 
 A simple proposal based on our existing implementation is that, we can add two new message types: `SnapshotExtensionMeta` 
 and `SnapshotExtensionPayload`, and they are appended to the existing multi-store stream with `SnapshotExtensionMeta` 
-acting as a delimiter between extensions. As the chunk hashes should be able to ensure data integrity, we don't need a
-delimiter to mark the end of the snapshot stream.
+acting as a delimiter between extensions. Even for modules that maintain the data outside of the tree, for determinism we
+require that the hash of the external data should be posted in the IAVL tree. As the chunk hashes should be able to ensure 
+data integrity, we don't need a delimiter to mark the end of the snapshot stream.
```

Comment:
I think the purpose of post external state hash into the merkle tree is to commit the external state in the app hash/header hash.

Reply:
so the chunk hashes should be something different from the hash in the merkle tree?

Reply:
what do you mean by `, we don't need a delimiter to mark the end of the snapshot stream.`?

Reply:
I add this sentence to clarify the question from @ethanfrey `I do not see some delimiter to mark the end of said stream, which seems like it would be useful to handle errors. Difference between connection break and finalised stream.`

Reply:
> I add this sentence to clarify the question from @ethanfrey `I do not see some delimiter to mark the end of said stream, which seems like it would be useful to handle errors. Difference between connection break and finalised stream.`

We have chunk hashes to detect a corrupted stream, so there's no problem here I think.

Reply:
so chunk hashes should be something different from the hash in the merkle tree, right?

Reply:
> so chunk hashes should be something different from the hash in the merkle tree, right?

yes, it's different. chunk hashes are for data integrity check during transport, merkle tree hash is for on-chain consensus check.